nsq/async.py

```diff
@@ -468,9 +471,18 @@ def _on_data(self, data, **kwargs):
             self.in_flight += 1
 
             message = protocol.decode_message(data)
+            if self.msg_timeout:
+                if message.id in self.__timeouts:
```

Comment:
Shouldn't it be impossible to have a duplicate message in `self.__timeouts`?  In order to see the same message ID a message would need to have:

1. been explicitly requeued (in which case the timeout callback needs to be cleaned up)
2. timed out and been redelivered (in which case the timeout callback would have fired and been cleaned up)

So, I don't think we need this check?